Framework/DataHandling/inc/MantidDataHandling/LoadAsciiStl.h

```diff
@@ -11,7 +17,7 @@ class DLLExport LoadAsciiStl : LoadStl {
 public:
   LoadAsciiStl(std::string filename) : LoadStl(filename) {}
   std::unique_ptr<Geometry::MeshObject> readStl() override;
-  bool isAsciiSTL();
+  bool isAsciiSTL() const;
```

Comment:
👍 for changing to const, but the method is not a good one. I know you did not write this but should probably be fixed here.

The following operation would throw

```cpp
LoadAsciiSLT myAlg;
myAlg.isAsciiSTL();
```

As `m_filename` not a valid path.

Would be best to make this a static or free function. Explicitly pass the `filename` as an argument so that you don't get tangled up int the rest of the state lifetime of the `Algorithm`. And also add a test for this functionality.

Reply:
since the point of this is simply to check that its ready to read an STL file, wouldn't it make more sense to simply return false if it doesn't have a valid filename?

Reply:
Yes I agree. I still don't like it being "attached" to `LoadAciiSLT`, because you don't have the filename as part of the constructor for that algorithm, so there is temporal coupling between calling `setProperty` and calling this `isAsciiSTL`?

Reply:
`LoadAsciiSTL` isn't actually an algorithm in its own right, Its just a class that provides functionality used by both the `LoadSampleShape` and `LoadSampleEnvironment`algorithms